CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/io/DraggableCellNGTest.java

```diff
@@ -0,0 +1,241 @@
+            transferModes.add(TransferMode.MOVE);
+            transferModes.add(TransferMode.LINK);
+            when(dragboard.getTransferModes()).thenReturn(transferModes);
+            
+            // NOTE: transfer mode set to something other than move to demonstrate test doesn't require event to to be move mode
+            final DragEvent event  = new DragEvent(null, dragboard, 0, 0, 0, 0, TransferMode.COPY, null, null, null);
+            
+            instance.getOnDragOver().handle(event);
+            
+            assertEquals(event.isConsumed(), true);
+            assertEquals(event.getAcceptedTransferMode(), TransferMode.MOVE);
+        }
+    }
+    
+    /**
+     * Test of onDragEntered Listener
+     */
+    @Test
+    public void testOnDragEntered() {
+        System.out.println("onDragEntered");
+        
+        // TODO: Find a way to instantiate toolkit in a headless environment
+        // This unit test should pass locally but will hold up in a headless environment (e.g. CI)
+        // Putting it in this if loop is a temp fix for the hold up in CI but it does it by skipping the test 
+        // (so CI won't actually tell you whether this is passing or not)
+        if (!GraphicsEnvironment.isHeadless()) {
+            new JFXPanel();
+            
+            final DraggableCell<String> instance = new DraggableCell<>();
+            
+            // mocking the required dragboard
+            final Dragboard dragboard = mock(Dragboard.class);
+            when(dragboard.hasString()).thenReturn(true);
+            
+            final DragEvent event  = new DragEvent(null, dragboard, 0, 0, 0, 0, null, null, null, null);
+            
+            instance.getOnDragEntered().handle(event);
+            
+            assertEquals(instance.getOpacity(), 0.3);
+        }
+    }
+    
+    /**
+     * Test of onDragExited Listener
+     */
+    @Test
+    public void testOnDragExited() {
+        System.out.println("onDragExited");
+        
+        // TODO: Find a way to instantiate toolkit in a headless environment
+        // This unit test should pass locally but will hold up in a headless environment (e.g. CI)
+        // Putting it in this if loop is a temp fix for the hold up in CI but it does it by skipping the test 
+        // (so CI won't actually tell you whether this is passing or not)
+        if (!GraphicsEnvironment.isHeadless()) {
+            new JFXPanel();
+            
+            final DraggableCell<String> instance = new DraggableCell<>();
+            
+            // mocking the required dragboard
+            final Dragboard dragboard = mock(Dragboard.class);
+            when(dragboard.hasString()).thenReturn(true);
+            
+            final DragEvent event  = new DragEvent(null, dragboard, 0, 0, 0, 0, null, null, null, null);
+            
+            instance.getOnDragExited().handle(event);
+            
+            assertEquals(instance.getOpacity(), 1.0);
```

Comment:
Would it be beneficial to have a call to one of the other tests to set the opacity to something other than 1.0, so that you can verify that it has actually changed between events and not just sitting at its default value?

ie. 
```
setOpacity(0.3);
assertEquals(getOpacity(), 0.3f);
doOnDrag();
assertEquals(getOpacity(), 1.0f);
```

Reply:
Good pickup. I'll add that